### PR TITLE
Fix #1735: Add Java Linter to the project

### LINKED ISSUE #1735: Add Java Linter to the project
With the Bazel migration, we are now having java files in our codebase. We would like to add a linter that ensures all the Java files follow the same style.

- [ ] Run the linter on our existing Java files. Create a PR that fixes all lint problems that the linter catches

- [ ] Add steps in the linter job of our GitHub Action workflow (similar to the [Kotlin linting](https://github.com/oppia/oppia-android/blob/develop/.github/workflows/main.yml#L27) step) that will install and run the linter on our Java files

- [ ] Verify that the Android Studio auto-formatter is using the same rules as the linter configuration. If not, make changes to the code style of the Java files in Android Studio (that will create changes to the files present in the .idea/codeStyles/ directory). Create a PR that checks in the changes in these files.

- [ ] Add documentation on how to install the linter in the [prerequisites section of our wiki](https://github.com/oppia/oppia-android/wiki#prerequisites)

### PR DESCRIPTION
**Pull request by anandwana001**
## Explanation

Fix #1735 

## For Reviewers
1. Download .jar file from here - https://github.com/checkstyle/checkstyle/releases/download/checkstyle-8.37/checkstyle-8.37-all.jar
2. Open terminal
3. Move to the path where the `.jar` file is located. 
4. Run the command `java -jar checkstyle-8.37-all.jar -c=checkstyle.xml /ABSOLUTE_PATH_TO_OPPIA_ANDROID/app/src/ /ABSOLUTE_PATH_TO_OPPIA_ANDROID/data/src/ /ABSOLUTE_PATH_TO_OPPIA_ANDROID/domain/src/ /ABSOLUTE_PATH_TO_OPPIA_ANDROID/utility/src/ /ABSOLUTE_PATH_TO_OPPIA_ANDROID/testing/src/`

Once this PR gets merge, I will add a new guide page in our wiki. 

Screenshot 2020-11-11 at 08 24 00

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: .github/workflows/main.yml
Comment (anandwana001) [inline .github/workflows/main.yml]: @vinitamurthi 
command should be 
```console
foo@bar:~$ 
java -jar checkstyle-8.37-all.jar -c=checkstyle.xml 
app/src/main/java/org/oppia/android/app/databinding/DrawableBindingAdapters.java
```

Comment (anandwana001) [reply .github/workflows/main.yml]: Done, looks like it is working now, but it should fail on these warning!! Checking into this

Screenshot 2020-11-10 at 16 03 23

Comment (anandwana001) [reply .github/workflows/main.yml]: on local machine, 

```console
java --version
java 10.0.2 2018-07-17
Java(TM) SE Runtime Environment 18.3 (build 10.0.2+13)
Java HotSpot(TM) 64-Bit Server VM 18.3 (build 10.0.2+13, mixed mode)
```

```console
Starting audit...
[WARN] DrawableBindingAdapters.java:22:3: 'method def rcurly' has incorrect indentation level 2, expected level should be 4. [Indentation]
[WARN] DrawableBindingAdapters.java:24:3: Missing a Javadoc comment. [MissingJavadocMethod]
[WARN] DrawableBindingAdapters.java:30:3: 'method def rcurly' has incorrect indentation level 2, expected level should be 4. [Indentation]
[WARN] DrawableBindingAdapters.java:32:3: Missing a Javadoc comment. [MissingJavadocMethod]
[WARN] DrawableBindingAdapters.java:38:3: 'method def rcurly' has incorrect indentation level 2, expected level should be 4. [Indentation]
[WARN] DrawableBindingAdapters.java:40:3: Missing a Javadoc comment. [MissingJavadocMethod]
[WARN] DrawableBindingAdapters.java:46:3: 'method def rcurly' has incorrect indentation level 2, expected level should be 4. [Indentation]
[WARN] DrawableBindingAdapters.java:51:3: 'method def rcurly' has incorrect indentation level 2, expected level should be 4. [Indentation]
[WARN] DrawableBindingAdapters.java:52:1: 'class def rcurly' has incorrect indentation level 0, expected level should be 2. [Indentation]
Audit done.
```

Comment (anandwana001) [reply .github/workflows/main.yml]: updating java version not worked

Comment (anandwana001) [reply .github/workflows/main.yml]: From our `Project.xml`, the indent is 2 but warning required 4 as per google java style.

```xml
<codeStyleSettings language="JAVA">
      <indentOptions>
        <option name="INDENT_SIZE" value="2" />
        <option name="CONTINUATION_INDENT_SIZE" value="4" />
        <option name="TAB_SIZE" value="2" />
      </indentOptions>
    </codeStyleSettings>
```

Is there any way to update project.xml and reflect the same in all files?

File: checkstyle.xml
Comment (anandwana001) [inline checkstyle.xml]: will delete this file, as we are using the jar directly and `/google_checks.xml` from it for configuration.

File: .github/workflows/main.yml
Comment (anandwana001) [inline .github/workflows/main.yml]: added only one file as of now to check the errors and warning reflected in the checks

File: .github/workflows/main.yml
Comment (anandwana001) [reply .github/workflows/main.yml]: Screenshot 2020-11-10 at 16 40 04

Comment (anandwana001) [reply .github/workflows/main.yml]: the pattern is, if the warning is something related to **rcurly** , it's not coming in checks
1. 'method def rcurly'
2. 'class def rcurly'

Comment (anandwana001) [reply .github/workflows/main.yml]: Screenshot 2020-11-10 at 17 29 26

File: checkstyle.xml
Comment (anandwana001) [reply checkstyle.xml]: No need to delete this file, as earlier we are using `/google_checks.xml` but it is not possible to update configuration in this file and we need error to be failed, so we have a copy of this file as `checkstyle.xml`

File: .github/workflows/main.yml
Comment (anandwana001) [inline .github/workflows/main.yml]: @vinitamurthi Need one more help here

failing with 
```console
Run java -jar checkstyle-8.37-all.jar -c=checkstyle.xml app/src/**/*.java && echo "Lint completed successfully"
Files to process must be specified, found 0.
Error: Process completed with exit code 255.
```

File: .github/workflows/main.yml
Comment (vinitamurthi) [inline .github/workflows/main.yml]: Change to `# linters: Java, Kotlin, Protobuf`
(Keep them alphabetical)

File: .github/workflows/main.yml
Comment (vinitamurthi) [inline .github/workflows/main.yml]: Restructure these linters to run in the order mentioned above - Java, Kotlin, Protobuf

File: app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java
Comment (vinitamurthi) [inline app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java:61]: Oh yikes, how was this working so far !

Comment (vinitamurthi) [reply app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java:61]: This is the expected behavior right? There should be no fallthrough?

File: checkstyle.xml
Comment (vinitamurthi) [inline checkstyle.xml]: I am a bit conflicted about this comment. I understand we have picked up the google style xml file but should we be adding their comment or should we add our own and mention that we have picked it up from this particular xml file and reference it? @seanlip can you comment on this - this is more of a open source question?

File: checkstyle.xml
Comment (vinitamurthi) [inline checkstyle.xml]: Does this checker even do XML layout files?

File: checkstyle.xml
Comment (seanlip) [reply checkstyle.xml]: Could you or @anandwana001 please point me to the repo where this code snippet was taken from, and link to the original file from which these lines were taken? I think I need that info in order to weigh in.

Comment (anandwana001) [reply checkstyle.xml]: @seanlip From here - https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml

Comment (seanlip) [reply checkstyle.xml]: OK, I did some reading and I think it would be better not to include an LGPL file as part of an Apache repo. Can you write a custom XML checkstyle for Oppia Android rather than using the default?

Also, in general, please never submit any code that's copied verbatim from other places to this repo. All code submitted in PRs must be original (including XML files).

File: app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java:61]: mentioning @rt4914 to confirm this is the expected behaviour.

File: checkstyle.xml
Comment (anandwana001) [reply checkstyle.xml]: It doesn't work with Android XML layout file.

File: app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java
Comment (rt4914) [reply app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java:61]: @vinitamurthi @anandwana001 
`break` should be there and even I am wondering how is this working correctly in develop.

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java
Comment (rt4914) [inline app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]: Nit:
We can write this in single line too. Right?
`/** BindingAdapter to set Rotation Animation and Rotation Angle. */`

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL @anandwana001

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]: ah, missed it. I will make a consistency here in all.

File: .github/workflows/main.yml
Comment (anandwana001) [reply .github/workflows/main.yml]: done

File: .github/workflows/main.yml
Comment (anandwana001) [reply .github/workflows/main.yml]: done

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]: done

File: app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java
Comment (rt4914) [reply app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java:61]: Screenshot_1605154019

After having discussing with @anandwana001 we found that it is not working in walkthrough and is getting used only there.

File: checkstyle.xml
Comment (anandwana001) [reply checkstyle.xml]: @seanlip  This checkstyle file (Google Java Style Guide) is something we need to follow as per Ben's comment - https://github.com/oppia/oppia-android/issues/1735#issuecomment-683076568 

If I create a custom, all I could do is remove comments, else all properties will remain the same as we want the same style to follow.

Comment (seanlip) [reply checkstyle.xml]: Discussed offline. @anandwana001 will be creating a checkstyle file suitable for Oppia Android's purposes from scratch.

Comment (anandwana001) [reply checkstyle.xml]: Adding @BenHenning for reference, 

I was planning to use the google style config directly from the command which is one of the recommended as per the documentation, but the default is only WARN where it will not fail/error on checks. So, I had created this config file which is same as the google java style, only changed the property called severity to error so that checks fail. 

[discussed with the repo owner](https://github.com/checkstyle/checkstyle/issues/4452#issuecomment-727278218) if you could change param in the command, but it's not possible so I will be writing this file from fresh.

File: app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java
Comment (BenHenning) [reply app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java:61]: I'm guessing that this was missed when the file was translated to Java. The original was probably something like:

```kotlin
return when (state) {
  VALUE1 -> result1
  VALUE2 -> result 2
  ...
}
```

I'm glad this was found.

File: checkstyle.xml
Comment (BenHenning) [reply checkstyle.xml]: It's probably preferable to have our own file, anyway, since it lets us ensure the enforce style exactly matches our style guide.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @anandwana001. Will take a pass on this once the style file is updated.

File: app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java
Comment (BenHenning) [reply app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java:61]: Also, @anandwana001 can you file an issue to track improving the code coverage here? We should have caught this in tests, but we apparently aren't covering these lines correctly.

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks. Keeping it blocked on final Ben's review.

**Review by vinitamurthi — APPROVED** [review]

**Comment by BenHenning** [comment]
@anandwana001 this is marked as draft--are you looking for specific feedback or is it ready for review?

**Comment by anandwana001** [comment]
> @anandwana001 this is marked as draft--are you looking for specific feedback or is it ready for review?

Just the feedback for the checkstyle.xml file.

File: checkstyle.xml
Comment (BenHenning) [inline checkstyle.xml]: Can you please include any files/resources you used in reference when creating this? Also, can you reply to this comment with some context on how it was created? E.g. did you write it manually, did you use a tool to generate it, etc.

**Comment by BenHenning** [comment]
Thanks @anandwana001. I lack some context which is making it difficult for me to spot check that file. Please respond to my comment & assign back to me--I'll take a look once I have a bit more context.

File: checkstyle.xml
Comment (anandwana001) [reply checkstyle.xml]: I had taken the reference from the original file 
https://github.com/checkstyle/checkstyle/blob/master/src/main/resources/google_checks.xml

As we need to follow the Java Google format, we need all the checks from the google_checks.xml

File: .github/workflows/main.yml
Comment (BenHenning) [inline .github/workflows/main.yml:35]: Just to check, is this needed for Checkstyle? I suggest adding a comment explaining that since we usually default to JDK 8 because of Android.

File: app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java
Comment (BenHenning) [reply app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java:61]: Reminder: please resolve this comment.

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]:
```suggestion
  /** Adapter to animate-rotate a view by the specified angle in the specified direction. */
```

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]:
```suggestion
  /** Adapter to set the height of a view. */
```

File: checkstyle.xml
Comment (BenHenning) [reply checkstyle.xml]: Given that there's not really a reasonable way to derive our own version of the file without it more or less being a copy of the original, the easiest solution available to us seems to be to keep the default severity as warning rather than error. Given also that we plan to only keep Java around temporarily, I think we can live with this.

However one main thing we need to check: when Checkstyle fails with a warning, is the application exit code 0 or 1? If the latter, then CI will properly fail even if it's outputting warnings.

**Review by BenHenning — CHANGES_REQUESTED** [review]

File: checkstyle.xml
Comment (anandwana001) [reply checkstyle.xml]: With severity warning the CI will not fail.

File: .github/workflows/main.yml
Comment (anandwana001) [reply .github/workflows/main.yml:35]: yes, adding comments.

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]: done

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]: done

File: app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/databinding/ImageViewBindingAdapters.java:61]: Done - https://github.com/oppia/oppia-android/issues/2449

File: .github/workflows/main.yml
Comment (BenHenning) [inline .github/workflows/main.yml]:
```suggestion
    # Java 11 is specifically needed for Checkstyle.
```

Reworded, and changed casing for consistency.

File: checkstyle.xml
Comment (BenHenning) [reply checkstyle.xml]: Does Checkstyle have any output if it succeeds? If not, could we do something like:

```bash
#!/bin/bash
lint_results=$(java -jar checkstyle-8.37-all.jar -c=checkstyle.xml app/src/ data/src/ domain/src/ utility/src/ testing/src/ 2>&1)
lint_command_result=$?
echo $lint_results
if [ $lint_command_result -ne 0] || [ -z "$lint_results" ]; then
  # Assume any lint output or non-zero exit code is a failure.
  exit 1
else
  exit 0
fi
```

This assumes any output is a fail. If the linter has regular output, then we might need to specifically check for 'warning' in the message. It does require that we put the linter command in a script, but we should be doing that, anyway, to avoid duplication with the configuration file.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Left some thoughts about how we might be able to live with 'warnings' and treat them as errors without needing a custom checkstyle configuration.

File: scripts/checkstyle_lint_check_github_action.sh
Comment (anandwana001) [inline scripts/checkstyle_lint_check_github_action.sh]: @BenHenning creating this separate file because for pre-push we need `../oppia-android-tools/`.

To remove this duplicacy of the file, we can change the path of pre-push files!! But we can't put setup files inside oppia-android folder and oppia-android-tools folder is something local, not available for GitHub action!!

File: .github/workflows/main.yml
Comment (anandwana001) [reply .github/workflows/main.yml]: done

Comment (anandwana001) [reply .github/workflows/main.yml]: resolving on behalf of ben as this is a straight change.

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]: resolving on behalf of ben as this is a straight change.

File: app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java
Comment (anandwana001) [reply app/src/main/java/org/oppia/android/app/databinding/ViewBindingAdapters.java]: resolving on behalf of ben as this is a straight change

File: checkstyle.xml
Comment (anandwana001) [reply checkstyle.xml]: removed checkstyle file.

File: checkstyle.xml
Comment (anandwana001) [reply checkstyle.xml]: Done

File: scripts/checkstyle_lint_check_github_action.sh
Comment (anandwana001) [reply scripts/checkstyle_lint_check_github_action.sh]: Adding @Srikanth-Kb for any thoughts on this.

**Comment by srikanthkb** [comment]
Here's an update for executing this linter on Git Bash - Windows 10,
1. Executing `bash scripts/setup.sh` is successful - the `.jar` file is downloaded successfully.
2. On executing `bash scripts/checkstyle_lint_check.sh` , we get the following error:
image
3. So, I manually downloaded the `google_checks.xml` from https://github.com/checkstyle/checkstyle/blob/14005e371803bd52dff429904b354dc3e72638c0/src/main/resources/google_checks.xml
4. Copied it to the path `C:/Program Files/Git/google_checks.xml` (i.e. path to Git directory) 
5. Executed the command `bash scripts/checkstyle_lint_check.sh` , in this case the execution was successful.
image

File: .github/workflows/main.yml
Comment (BenHenning) [inline .github/workflows/main.yml:37]: Do we still need this if the script downloads it?

File: scripts/checkstyle_lint_check_github_action.sh
Comment (BenHenning) [reply scripts/checkstyle_lint_check_github_action.sh]: Hmm, I was expecting to have a single script rather than 2. Why doesn't oppia-android-tools work in the GitHub action world?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @anandwana001! This is looking close. My main comment left is around consolidating the two checkstyle scripts into one.

File: .github/workflows/main.yml
Comment (anandwana001) [reply .github/workflows/main.yml:37]: yes, right now, I cannot use our setup script here. 
Setup script is doing two things
1. copying the pre-push script to hooks
2. download all the required setup files for ktlint etc

Reason 1- point 1 is not needed in the GitHub action.
Reason 2- GitHub action is not understanding the `../oppia-android-tools`, so I had mentioned `$HOME` for now. Creating a separate issue for this where we can have a balanced use of pre-push scripts and GitHub action altogether.

Comment (BenHenning) [reply .github/workflows/main.yml:37]: Ideally we should try to fix that in this PR for the Java linter rather than checking in two places that both have separate linter versions.

I'm actually not suggesting that we run the setup script. Instead, could we move out the dependencies bits into a "setup_deps.sh" script & use that in actions, instead? Then "setup.sh" would call "setup_deps.sh" & set up the pre-push hook.

**Review by BenHenning — COMMENTED** [review]
Thanks @anandwana001. Just one comment.

File: scripts/checkstyle_lint_check_github_action.sh
Comment (BenHenning) [reply scripts/checkstyle_lint_check_github_action.sh]: Continuing this discussion. on the newer comment thread below.

File: checkstyle.xml
Comment (BenHenning) [reply checkstyle.xml]: Just to check, have you confirmed that this correctly fails in CI with missing Java style issues checked in?

**Comment by Luffy18346** [comment]
@anandwana001 PTAL

**Comment by anandwana001** [comment]
Will update this week.

**Comment by Luffy18346** [comment]
> Will update this week.

Ok

File: .github/workflows/main.yml
Comment (anandwana001) [reply .github/workflows/main.yml]: resolving on behalf of Vinita

File: .github/workflows/main.yml
Comment (anandwana001) [reply .github/workflows/main.yml]: resolving on behalf of Vinita

**Comment by anandwana001** [comment]
@BenHenning I had modularize the scripts. Now the pre-push and actions both are using the same scripts.
Once this pr get's merged, I will apply the same to buf and bazel pr also.

File: .github/workflows/main.yml
Comment (anandwana001) [reply .github/workflows/main.yml:37]: Done, thanks for the suggestion.

**Comment by Luffy18346** [comment]
@BenHenning PTAL

File: scripts/setup.sh
Comment (BenHenning) [inline scripts/setup.sh]: This script intended to check that all dependencies were downloaded  not perform the hook setup. I think you can just remove setup_deps and keep the hook bits in here based on how you set things up.

File: checkstyle.xml
Comment (BenHenning) [reply checkstyle.xml]: @anandwana001 can you confirm this?

**Comment by BenHenning** [comment]
Thanks @anandwana001! I find this much cleaner. Just had 1 new comment & 1 old comment.

File: checkstyle.xml
Comment (anandwana001) [reply checkstyle.xml]: yes, it fails correctly.

File: scripts/setup.sh
Comment (anandwana001) [reply scripts/setup.sh]: Done

File: checkstyle.xml
Comment (anandwana001) [reply checkstyle.xml]: ah, corrected, I was using sh earlier. fixed it now.

Comment (BenHenning) [reply checkstyle.xml]: Thanks for checking!

**Review by BenHenning — APPROVED** [review]
Thanks @anandwana001! PR LGTM. Looks like the CI failure is a flake. Re-running tests.

**Comment by Pranav-Bobde** [comment]
error
image
image




Can anyone help me resolve this error...Please!

**Comment by anandwana001** [comment]
@Pranav-Bobde Have you used the correct xml file - https://github.com/checkstyle/checkstyle/blob/14005e371803bd52dff429904b354dc3e72638c0/src/main/resources/google_checks.xml ?

**Comment by anandwana001** [comment]
Let me know, if you follow the installation steps correctly - https://github.com/oppia/oppia-android/wiki#installation

**Comment by Pranav-Bobde** [comment]
@anandwana001 Yes, I have used the correct file and followed correct steps. In fact, i've deleted, re-forked and done all steps more than 4-5 times now.

**Comment by anandwana001** [comment]
Could you check the `google_checks.xml` file is correct and only the required file presented in the system, not the other version of this file in the system?

**Comment by Pranav-Bobde** [comment]
image
yup just one google_checks.xml here

**Comment by FareesHussain** [comment]
@Pranav-Bobde can you delete the current google_check.xml and add this file
https://github.com/checkstyle/checkstyle/blob/14005e371803bd52dff429904b354dc3e72638c0/src/main/resources/google_checks.xml

it is supposed to be 17KB size here

**Comment by anandwana001** [comment]
@FareesHussain We had faced this issue earlier, could you confirm is the correct file of 18KB?

**Comment by FareesHussain** [comment]
The correct files is of 17KB

**Comment by Pranav-Bobde** [comment]
@FareesHussain  @anandwana001  It's 17.1 Kb only . 
image

**Comment by anandwana001** [comment]
@Pranav-Bobde Could we confirm it's the same file from that particular we had mentioned in the link? 
https://github.com/checkstyle/checkstyle/blob/14005e371803bd52dff429904b354dc3e72638c0/src/main/resources/google_checks.xml

**Comment by Pranav-Bobde** [comment]
@anandwana001 
1) Gone to the link
1image
2) Gone to code and downloaded zip
image
3) Extracted & Pasted the file in the Git Directory
image

**Comment by FareesHussain** [comment]
can you try be deleting and adding the new one

**Comment by FareesHussain** [comment]
In my case, the file is 17.4KB

**Comment by Pranav-Bobde** [comment]
@FareesHussain 
Deleted and added. Nothing Changed.

**Comment by Pranav-Bobde** [comment]
@FareesHussain  @anandwana001 
Would you like to anydesk with me?

**Comment by anandwana001** [comment]
The issue is, you are downloading code and using the file from maste branch, not from that particular commit we had mentioned in the link.

**Comment by FareesHussain** [comment]
To download open this link https://raw.githubusercontent.com/checkstyle/checkstyle/14005e371803bd52dff429904b354dc3e72638c0/src/main/resources/google_checks.xml

and CTRL + D to download the file

**Comment by anandwana001** [comment]
Or you can simply right click on raw button and download the file.

**Comment by Pranav-Bobde** [comment]
@anandwana001 @FareesHussain 
Ok. It's working . I just copied the content and made a xml file from sublime.
TYSM for your help. Really Appreciate it.

**Comment by Pranav-Bobde** [comment]
@anandwana001  @FareesHussain  
Can you please tell me where am i supposed to assign the assignee & reviewer?
image
Because i think it should've been over here.
But i'm not getting that section anywhere on my screen.

**Pull request merged**
